### ISSUE TITLE
Cannot use escape string in for-in array

### ISSUE THREAD

**Issue by morishin** [google/jsonnet#791]
It looks like bug for me 🤔 

NG
![スクリーンショット 2020-04-03 18 28 41](https://user-images.githubusercontent.com/1413408/78345825-23f78b00-75d9-11ea-9a00-986d9b015996.png)

OK
![スクリーンショット 2020-04-03 18 28 28](https://user-images.githubusercontent.com/1413408/78345829-2659e500-75d9-11ea-993f-ea4ccdbe10dc.png)

**Comment by sparkprime** [comment]
Yeah, nice find.  It's a bug in the desugarer which handles the escapes but only in the C++ version.  Go version is fine.  Javascript is derived from C++ so that's why you see it on the website.

**Comment by sparkprime** [comment]
It actually desugars it twice, that's the problem.
The first time here: https://github.com/google/jsonnet/blob/master/core/desugarer.cpp#L628
And then it embeds the spec in a new array comprehension and desugars the whole thing including the specs here: https://github.com/google/jsonnet/blob/master/core/desugarer.cpp#L669

**Comment by morishin** [comment]
Thank you!
